src/engine/SCons/Script/Main.py

```diff
@@ -1099,17 +1100,36 @@ def _main(parser):
                                           targets, target_top)
 
     else:
-        # Write the build graph
-        graph = os.environ.get("SCONS_GRAPH")
-        if graph:
+        scons_mode = os.environ.get("SCONS_MODE")
+        if options.mode == "vulcan" or scons_mode == "vulcan":
```

Comment:
If I set `--mode=local` on the command line and `SCONS_MODE=vulcan` in the environment, this code operates in vulcan mode. That's the wrong behavior.